pennylane_lightning/lightning_kokkos/_state_vector.py

```diff
@@ -0,0 +1,390 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Class implementation for lightning_kokkos state-vector manipulation.
+"""
+
+try:
+    from pennylane_lightning.lightning_kokkos_ops import (
+        InitializationSettings,
+        StateVectorC64,
+        StateVectorC128,
+        allocate_aligned_array,
+        print_configuration,
+    )
+except ImportError:
+    pass
+
+import numpy as np
+import pennylane as qml
+from pennylane import BasisState, DeviceError, StatePrep
+from pennylane.measurements import MidMeasureMP
+from pennylane.ops import Conditional
+from pennylane.ops.op_math import Adjoint
+from pennylane.tape import QuantumScript
+from pennylane.wires import Wires
+
+# pylint: disable=import-error, no-name-in-module, ungrouped-imports
+from pennylane_lightning.core._serialize import global_phase_diagonal
+
+from ._measurements import LightningKokkosMeasurements
+
+
+class LightningKokkosStateVector:  # pylint: disable=too-few-public-methods
```

Comment:
This class follows the same recipe as `Lightning Qubit ` with an exception for Native Controlled Gates